### PR TITLE
feature: Add Go client-side tools aligncheck and deadcode

### PR DESCRIPTION
**Pull request by pauloribeiro-codacy**
Adds instructions on how to run the new Go client-side tools aligncheck ([CY-3143](https://codacy.atlassian.net/browse/CY-3143)) and deadcode ([CY-3141](https://codacy.atlassian.net/browse/CY-3141)).

In this pull request, I'm also refactoring the way the documentation for the client-side tools is organized ([DOCS-147](https://codacy.atlassian.net/browse/DOCS-147)). Before, the only client-side tool that needed the Codacy Analysis CLI was SpotBugs. Since now more tools will require the Codacy Analysis CLI in the same way, we need to make the instructions more generic.

### CODE REVIEW

**Comment by pauloribeiro-codacy** [comment]
For reference, the following pull requests are related to the current one:

- https://github.com/codacy/codacy-gosec/pull/15
- https://github.com/codacy/codacy-spotbugs/pull/22

**Comment by pauloribeiro-codacy** [comment]
To make reviewing easier, here are the previews of the main pages changed or created on this pull request:

- [Client-side tools](https://github.com/codacy/docs/blob/feature/go-client-side-tools-CY-3254/docs/related-tools/local-analysis/client-side-tools.md)
- [Running aligncheck](https://github.com/codacy/docs/blob/feature/go-client-side-tools-CY-3254/docs/related-tools/local-analysis/running-aligncheck.md)
- [Running deadcode](https://github.com/codacy/docs/blob/feature/go-client-side-tools-CY-3254/docs/related-tools/local-analysis/running-deadcode.md)
- [Running SpotBugs](https://github.com/codacy/docs/blob/feature/go-client-side-tools-CY-3254/docs/related-tools/local-analysis/running-spotbugs.md)

Currently, the page **Client-side tools** explains the concepts involved in client-side tools and acts as an index containing links to the instructions on how to run each client-side tool.

The instructions for "containerized tools" are quite similar; however, since there are also minor differences between them, taken together with a different process for "standalone tools", I couldn't figure out a solution to avoid the duplication without compromising the usability of the documentation for the end-users. So, for now, I prefer to have the duplication and I will implement a solution to [reuse content](https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins#reusing-content-snippets--includes) when I'm refactoring this whole area of the documentation.

File: docs/related-tools/local-analysis/client-side-tools.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Client-side tools
+
+Client-side tools enable you to run analysis either locally or as part of your CI process and integrating the results into your Codacy workflow. This way, Codacy presents the analysis information reported by your local tools alongside all other code quality information on the Codacy dashboards.
+
+Codacy supports client-side tools in two ways:
+
+-   **Standalone tools**
+
+    Codacy provides auxiliary converters that parse the output of third-party tools, convert the output format, and upload the results to Codacy. You must download, configure, and run the third-party tools yourself.
```
Comment (josemiguelmelo) [inline docs/related-tools/local-analysis/client-side-tools.md:9]: Standalone tools just convert the output and prints it to stdout. The upload part is a manual step that should be done by the user. Maybe we should rephrase this to something like:

```suggestion
    Codacy provides auxiliary converters that parse the output of third-party tools and converts it to its format, which can then be uploaded to Codacy. You must download, configure, and run the third-party tools yourself.
```

File: docs/faq/troubleshooting/why-is-my-file-missing.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ As a solution to this problem, we placed a size limit to the files that Codacy w
 
 ### What if I need to analyze a file that exceeds this limit?
 
-While Codacy will discard your file by default, you can still have it analyzed using the [Codacy Analysis CLI](../../related-tools/running-local-analysis.md) and running the analysis locally. CLI doesn't have a set timeout or any limitation on file size, so you will be able to have your files analyzed this way.
+While Codacy will discard your file by default, you can still have it analyzed using the [Codacy Analysis CLI](../../related-tools/local-analysis/running-local-analysis.md) and running the analysis locally. CLI doesn't have a set timeout or any limitation on file size, so you will be able to have your files analyzed this way.
```
Comment (pedrocodacy) [inline docs/faq/troubleshooting/why-is-my-file-missing.md:37]: `CLI doesn't have a set timeout or any limitation on file size` -> this is not 100% true - we do not have a timeout for the total analysis, but we do have a timeout for each tool execution. Maybe we can be clearer here on this or just not refer timeouts

File: docs/related-tools/local-analysis/client-side-tools.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    You can't configure these tools on the Codacy UI, since you manage their configuration locally.
+
+-   **Containerized tools**
```
Comment (franciscodua) [inline docs/related-tools/local-analysis/client-side-tools.md:13]: Which tools are containerized and which are not?

Comment (pauloribeiro-codacy) [reply docs/related-tools/local-analysis/client-side-tools.md:13]: Currently, I list all tools with an indication if they are "standalone" or "containerized" at the bottom of the page.

I did experiment with having separate lists for the two types of tools. However, I figured that customers will not choose the tools they will run based on their type, but depending on the programming languages they're using and their needs. Because of this, I decided to provide a simple list with all the tools (pointing to self-contained instructions on how to run them) to make the choice and setup of the tools as transparent as possible.

File: docs/related-tools/local-analysis/client-side-tools.md (hunk 1 above)
Comment (pauloribeiro-codacy) [reply docs/related-tools/local-analysis/client-side-tools.md]: Thanks @josemiguelmelo, this is an important detail that also helps to clarify the difference between the two types of tools.

File: docs/faq/troubleshooting/why-is-my-file-missing.md (hunk 2 above)
Comment (pauloribeiro-codacy) [reply docs/faq/troubleshooting/why-is-my-file-missing.md]: Well spotted @pedrocodacy! :mag: 

I removed the reference to the timeout since, in this context, it's precisely the time of each tool execution that could prevent the analysis from being successful. :thinking:

**Comment by pauloribeiro-codacy** [comment]
I forgot to mention that with this change, I'm grouping all documentation related to "local analysis" under its own section on the sidebar tree:

image

File: docs/faq/troubleshooting/why-is-my-file-missing.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ As a solution to this problem, we placed a size limit to the files that Codacy w
 
 ### What if I need to analyze a file that exceeds this limit?
 
-While Codacy will discard your file by default, you can still have it analyzed using the [Codacy Analysis CLI](../../related-tools/running-local-analysis.md) and running the analysis locally. CLI doesn't have a set timeout or any limitation on file size, so you will be able to have your files analyzed this way.
+While Codacy will discard your file by default, you can still have it analyzed using the [Codacy Analysis CLI](../../related-tools/local-analysis/running-local-analysis.md) and running the analysis locally. CLI doesn't any limitation on file size, so you will be able to have your files analyzed this way.
```
Comment (josemiguelmelo) [inline docs/faq/troubleshooting/why-is-my-file-missing.md:37]:
```suggestion
While Codacy will discard your file by default, you can still have it analyzed using the [Codacy Analysis CLI](../../related-tools/local-analysis/running-local-analysis.md) and running the analysis locally. CLI doesn't have any limitation on file size, so you will be able to have your files analyzed this way.
```

**Review by josemiguelmelo — APPROVED** [review]

File: docs/related-tools/local-analysis/running-spotbugs.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+The Codacy Analysis CLI tries to find the compiled classes and map results to the source files automatically. If you use Maven, Gradle, or sbt the Codacy Analysis CLI also detects the default layouts automatically.
+
+If there is an issue with detection, you can configure these paths manually by adding a `.codacy.yml` [Codacy configuration file](../../repositories-configure/codacy-configuration-file.md) to the root of the repository:
+
+```yml
+---
+engines:
+  spotbugs:
+    enabled: true
```
Comment (franciscodua) [inline docs/related-tools/local-analysis/running-spotbugs.md:55]: We have to remove this line. We don't support `enabled` for engines.

Comment (pauloribeiro-codacy) [reply docs/related-tools/local-analysis/running-spotbugs.md]: Thanks @franciscodua, I just removed the line with `enabled: true`. Please check.

**Review by franciscodua — APPROVED** [review]

**Review by IGFCoimbra — APPROVED** [review]
